File: C/FUNCIONAL/recursiva.c
#include <stdio.h>                              // Librería para entrada/salida

int factorial(int n) {                          // Función recursiva
    if (n == 0) return 1;                       // Caso base: si es 0, retorna 1
    return n * factorial(n - 1);                // Caso recursivo: n × factorial(n-1)
}

int main() {                                    // Función principal
    printf("%d\n", factorial(5));               // Calcular factorial de 5
    return 0;                                   // Fin del programa
}

File: JAVA/FUNCIONAL/recursiva.java
public class recursiva {                        // Clase principal
    static int factorial(int n) {               // Función recursiva
        if (n == 0) return 1;                   // Caso base: si es 0, retorna 1
        return n * factorial(n - 1);            // Caso recursivo: n × factorial(n-1)
    }
    
    public static void main(String[] args) {    // Método principal
        System.out.println(factorial(5));       // Calcular factorial de 5
    }
}
